src/modeler/AutoRest.Swagger/Validation/ServiceDefinitionParameters.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class ServiceDefinitionParameters : TypedRule<Dictionary<string, SwaggerParameter>>
+    {
+        private const string SubscriptionId = "subscriptionId";
+        private const string ApiVersion = "api-version";
+        /// <summary>
+        /// This rule passes if the parameters section contains both subscriptionId and api-version parameters
+        /// </summary>
+        /// <param name="paths"></param>
+        /// <returns></returns>
+        public override bool IsValid(Dictionary<string, SwaggerParameter> ParametersMap)
+        {
+            if (ParametersMap.Count==0 || ParametersMap == null)
+                return false;
+
+            var Names = ParametersMap.Values.Select(Param => Param.Name).ToList();
+            return Names.Contains(SubscriptionId) && Names.Contains(ApiVersion);
+        }
```

Comment:
Now, I get to the bottom and see that you could simplify it a bit more, and use a single expression that leverages null-propagation operator (which turns the expression result into a nullable bool result `bool?` which can be trivially coaxed back into a bool by adding  `true == ` at the front of the expression.

``` csharp
  public override bool IsValid(Dictionary<string, SwaggerParameter> ParametersMap) => true == ParametersMap?.Values.Any( parameter => parameter.Name == SubscriptionId  || parameter.Name == ApiVersion );
```